uwblueprint/paramedics-react
# Front end mutation cleanup

[Notion ticket](https://www.notion.so/uwblueprintexecs/delete-and-edit-mutation-cleanup-98ad81733016484b928909fb184e7360)

## Changes
For each entity type
- Move url navigation after add/edit to `useMutation`'s `onCompleted` callback
    - Use `history.replace()` instead of `history.push()`
- Move `enqueueSnackbar` calls to `useMutation`'s `onCompleted` callback
- Add missing snackbars (based on Figma)

Entity types
- Ambulance
- CCP
- Event
- Hospital
- Patient
- User

Additional scope (added to notion ticket):
- Remove option menu when the delete confirmation modal opens (should be fixed for the following)
    - Ambulance
    - CCP
    - Hospital
    - User

## Testing
For each of the listed entities above:
- Check add/edit/delete mutations
    - Confirm cache/UI updates correctly
    - Snackbars appear (if in Figma designs)
- Check the option menu is closed when the delete confirmation modal opens

Here are the urls for accessing the general flow for mutating each entity:
- Ambulance: `/manage/ambulances`
- CCP: `/events/{ccpId}`
- Event: `/events`
- Hospital: `/manage/hospitals`
- Patient: `/events/{eventId}/ccps/{ccpId}`
- User (also referred to as team member): `/manage/members`

## Screenshots
Ambulance
![image](https://user-images.githubusercontent.com/22457859/96352460-4478c580-1091-11eb-9b78-e27cce261a0d.png)

CCP
![image](https://user-images.githubusercontent.com/22457859/96352389-c3b9c980-1090-11eb-8e2d-65656f1b5fb0.png)

Hospital
![image](https://user-images.githubusercontent.com/22457859/96352539-e9939e00-1091-11eb-9bb9-21053e6c83ca.png)

User
![image](https://user-images.githubusercontent.com/22457859/96352582-3bd4bf00-1092-11eb-8cbd-c39468c87e3b.png)


## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
